src/peer.ts

```diff
@@ -0,0 +1,44 @@
+import net from "net";
+
+export interface PeerOpts {
+  ip: string;
+  port: number;
+}
+
+export default class Peer {
+  public ip: string;
+  public port: number;
+  public connection: net.Socket;
+
+  /**
+   * Constructor
+   * @param {PeerOpts} opts
+   */
+  public constructor(opts: PeerOpts) {
+    this.ip = opts.ip;
+    this.port = opts.port;
+  }
+
+  /**
+   * Establishes a connection to the peer.
+   * @returns {boolean}
+   */
+  public connect = (): boolean => {
+    // Attempt to connect to peer, if connection refused remove the peer from bootnodes.
+    try {
+      this.connection = net.createConnection({port: this.port});
```

Comment:
consider making this function async, and awaiting a promisified version of createConnection, so we can wait until the connection actually connects.
eg: `this.connection = await promisify(net.createConnection)({port: this.port});`

Reply:
what are your thoughts on promisfy over classic
 async/await?

Reply:
You meant promisify over promises?

Promises can lead to callback hell and it's hard to pass results from it.
I think here error will never be detected since it will happen async. I think you need to do something like this:
```javascript
const that = this;
return new Promise((resolve, reject) => {
   that.connection = net.createConnection({port: this.port});
   that.connection.on('connect', resolve);
   that.connection.on('error', reject);
})
```
On a side note, I think you forgot to pass ip as host option